library/std/src/sys/fs/windows.rs

```diff
@@ -1140,6 +1141,18 @@ impl FileType {
     }
 }
 
+impl PartialEq for FileType {
+    fn eq(&self, other: &Self) -> bool {
+        self.is_directory() == other.is_directory() && self.is_symlink() == other.is_symlink()
+    }
+}
+
+impl Hash for FileType {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write(&[self.is_directory() as u8, self.is_symlink() as u8]);
```

Comment:
This could be squished into a single byte but not sure if such a micro size optimisation is worth it.